UnattendResources/Logon.ps1

```diff
@@ -291,43 +292,44 @@ try
     if ($vmwareToolsPath) {
         Install-VMwareTools
     }
-    Run-CustomScript "RunBeforeCloudbaseInitInstall.ps1"
-    $Host.UI.RawUI.WindowTitle = "Installing Cloudbase-Init..."
+    if ($cloudbaseIniInstall){
+        Run-CustomScript "RunBeforeCloudbaseInitInstall.ps1"
+        $Host.UI.RawUI.WindowTitle = "Installing Cloudbase-Init..."
 
-    $programFilesDir = $ENV:ProgramFiles
+        $programFilesDir = $ENV:ProgramFiles
 
-    $CloudbaseInitMsiPath = "$resourcesDir\CloudbaseInit.msi"
-    $CloudbaseInitMsiLog = "$resourcesDir\CloudbaseInit.log"
+        $CloudbaseInitMsiPath = "$resourcesDir\CloudbaseInit.msi"
+        $CloudbaseInitMsiLog = "$resourcesDir\CloudbaseInit.log"
 
-    if (!$serialPortName) {
-        $serialPorts = Get-WmiObject Win32_SerialPort
-        if ($serialPorts) {
+        if (!$serialPortName) {
+            $serialPorts = Get-WmiObject Win32_SerialPort
+            if ($serialPorts) {
             $serialPortName = $serialPorts[0].DeviceID
+            }
         }
-    }
-
-    $msiexecArgumentList = "/i $CloudbaseInitMsiPath /qn /l*v $CloudbaseInitMsiLog"
-    if ($serialPortName) {
-        $msiexecArgumentList += " LOGGINGSERIALPORTNAME=$serialPortName"
-    }
 
-    $p = Start-Process -Wait -PassThru -FilePath msiexec -ArgumentList $msiexecArgumentList
-    if ($p.ExitCode -ne 0) {
-        throw "Installing $CloudbaseInitMsiPath failed. Log: $CloudbaseInitMsiLog"
-    }
+        $msiexecArgumentList = "/i $CloudbaseInitMsiPath /qn /l*v $CloudbaseInitMsiLog"
+        if ($serialPortName) {
+            $msiexecArgumentList += " LOGGINGSERIALPORTNAME=$serialPortName"
+        }
 
-    $Host.UI.RawUI.WindowTitle = "Running SetSetupComplete..."
-    & "$programFilesDir\Cloudbase Solutions\Cloudbase-Init\bin\SetSetupComplete.cmd"
-    Run-CustomScript "RunAfterCloudbaseInitInstall.ps1"
+        $p = Start-Process -Wait -PassThru -FilePath msiexec -ArgumentList $msiexecArgumentList
+        if ($p.ExitCode -ne 0) {
+            throw "Installing $CloudbaseInitMsiPath failed. Log: $CloudbaseInitMsiLog"
+        }
 
-    Run-Defragment
+        $Host.UI.RawUI.WindowTitle = "Running SetSetupComplete..."
+        & "$programFilesDir\Cloudbase Solutions\Cloudbase-Init\bin\SetSetupComplete.cmd"
+        Run-CustomScript "RunAfterCloudbaseInitInstall.ps1"
 
-    Release-IP
+        Run-Defragment
 
-    $Host.UI.RawUI.WindowTitle = "Running Sysprep..."
-    $unattendedXmlPath = "$programFilesDir\Cloudbase Solutions\Cloudbase-Init\conf\Unattend.xml"
-    Set-PersistDrivers -Path $unattendedXmlPath -Persist:$persistDrivers
+        Release-IP
 
+        $Host.UI.RawUI.WindowTitle = "Running Sysprep..."
+        $unattendedXmlPath = "$programFilesDir\Cloudbase Solutions\Cloudbase-Init\conf\Unattend.xml"
+        Set-PersistDrivers -Path $unattendedXmlPath -Persist:$persistDrivers
+    }

```

Comment:
if you do not have cloudbase-init and there is no sysprep unattend.xml configured, sysprep should not be performed.